darts/tests/models/forecasting/test_classifier_model.py

```diff
@@ -0,0 +1,713 @@
+    sine_univariate3_cat = sine_univariate3.map(lambda x: np.round(x))
+
+    sine_multivariate1_cat = sine_univariate1_cat.stack(sine_univariate2_cat)
+    sine_multivariate2 = sine_univariate2.stack(sine_univariate3)
+
+    sine_multiseries1_cat = [
+        sine_univariate1_cat,
+        sine_univariate2_cat,
+        sine_univariate3_cat,
+    ]
+    sine_multiseries2 = [sine_univariate4, sine_univariate5, sine_univariate6]
+
+    probabilistic_classifiers = [
+        (LogisticRegression, {}),
+        (KNeighborsClassifier, {"n_neighbors": 3}),
+        (SVC, {"gamma": 2, "C": 1, "random_state": 42, "probability": True}),
+        (GaussianProcessClassifier, {"kernel": 1.0 * RBF(1.0), "random_state": 42}),
+        (DecisionTreeClassifier, {"max_depth": 5, "random_state": 42}),
+        (
+            RandomForestClassifier,
+            {"max_depth": 5, "n_estimators": 10, "max_features": 1, "random_state": 42},
+        ),
+        (MLPClassifier, {"alpha": 1, "max_iter": 1000, "random_state": 42}),
+        (AdaBoostClassifier, {"random_state": 42}),
+        (GaussianNB, {}),
+        (
+            XGBClassifierModel,
+            {
+                "n_estimators": 1,
+                "max_depth": 1,
+                "max_leaves": 1,
+                "random_state": 42,
+            },
+        ),
+    ]
+
+    if lgbm_available:
+        probabilistic_classifiers.append((
+            LightGBMClassifierModel,
+            {
+                "n_estimators": 1,
+                "max_depth": 1,
+                "num_leaves": 2,
+                "verbosity": -1,
+                "random_state": 42,
+            },
+        ))
+
+    if cb_available:
+        probabilistic_classifiers.append((
+            CatBoostClassifierModel,
+            {
+                "iterations": 1,
+                "depth": 1,
+                "verbose": -1,
+                "random_state": 42,
+            },
+        ))
+
+    @pytest.mark.parametrize(
+        "clf_params",
+        process_model_list(probabilistic_classifiers),
+    )
+    def test_class_proba_likelihood_median_pred_is_same_than_no_likelihood(
+        self, clf_params
+    ):
+        clf, kwargs = clf_params
+        model = clf(lags=2, **kwargs, likelihood=None)
+        model_likelihood = clf(
+            lags=2,
+            **kwargs,
+        )
+
+        model.fit(self.sine_univariate1_cat)
+        # model has no likelihood
+        with pytest.raises(ValueError) as err:
+            model.predict(2, predict_likelihood_parameters=True)
+        print(err.value)
+        assert (
+            str(err.value) == "`predict_likelihood_parameters=True` is only"
+            " supported for probabilistic models fitted with a likelihood."
+        )
+
+        model_likelihood.fit(self.sine_univariate1_cat)
+        # model_likelihood has ClassProbability likelihood
+        probas = model_likelihood.predict(1, predict_likelihood_parameters=True)
```

Comment:
Nice :) 

We should add a couple more checks. You can use `darts.tests.models.forecasting.test_sf_models.TestSFModels.test_probabilistic_support` as a reference for what we should check (reproducibility, correct parameter names for parameter predictions, accuracy, sampled vs likelihood parameter vs median foreacst, ...).